01/highest-rated.md

```diff
@@ -0,0 +1,183 @@
+# Highest Rated
+
+## Problem Statement
+
+Imagine working on software that deals with restaurant reviews.
+
+Create a function named `get_highest_rated` that is responsible for finding the highest-rated restaurant.
+
+This function should take in a list of dictionaries named `restaurants` as a parameter. Each dictionary represents the data associated with a restaurant, including its rating. This function should have a return value of the restaurant with the highest rating.
+
+### Example Input/Output
+
+|<div style="width:250px">Input (Argument of the function)</div>|<div style="width:250px">Expected Output <br> (Return value of the function)</div> |<div style="width:600px">Explanation</div> |
+| -- | -- | --|
+| `restaurants = [{"name": "Grillby's", "rating": 1}, {"name": "Crow's Nest", "rating": 5}]` | `{"name": "Crow's Nest", "rating": 5}`         | There are two restaurants: "Grillby's" and "Crow's Nest." The rating of "Grillby's" is `1`, and the rating of "Crow's Nest" is `5`. "Crow's Nest" has the highest rating. The return value of this function should be a dictionary, which contains the name and rating of this restaurant. |
+| `restaurants = [{"name": "Crow's Nest", "rating": 1}]` | `{"name": "Crow's Nest", "rating": 1}`         | There is one restaurant in the input. Its name is "Crow's Nest," and its rating is `1`. Even though there's only one restaurant, it has the highest rating in this list! The return value of this function should be a dictionary, which contains the name and rating of "Crow's Nest." |
+| `restaurants = []`                              | `None`                                         | The input is an empty list, which means that there are zero restaurants. Because there are _no_ restaurants at all, the output, or return value, of this function should be `None`. |
```

Comment:
Maybe we leave off this line to provide some clarifying questions as well as ties.

```suggestion
```

Reply:
I was thinking in the future we'd leave off this line, but for this first PSE if the test cases came straight from the example input/output, that would be okay. Open to leaving it off too, though.

Reply:
hmm I'm torn! I think leaving it for the first one would be good. Maybe even a note that this a common edge case to think about. Sometimes problems won't give you them all, and it's your job to think about these knowledge gaps

then they are prepared for future ambiguity??